src/OrchardCore.Modules/OrchardCore.Setup/Views/_Layout.cshtml

```diff
@@ -21,5 +42,16 @@
             @await RenderSectionAsync("Footer", required: false)
         </footer>
     </div>
+    <script>
+        function setLocalizationUrl() {
+            var culturesList = document.getElementById('culturesList');
+            var culture = culturesList.options[culturesList.selectedIndex].value
+            var queryString = window.location.search;
+            var searchParams = new URLSearchParams(queryString);
+
+            searchParams.set("ui-culture", culture);
+            window.location = window.location.origin + '@(!String.IsNullOrWhiteSpace(Context.Request.PathBase.Value) ? Context.Request.PathBase.Value : "")' + '?' + searchParams.toString();
```

Comment:
For infos `PathSting.HasValue` (here `PathBase.Value`) does the same so that's okay

        /// <summary>
        /// True if the path is not empty
        /// </summary>
        public bool HasValue
        {
            get { return !string.IsNullOrEmpty(_value); }
        }

There are some places where we fully encode all uri components, but yes there are some places where we know or want to work on an unencoded url string and where we use `Request.PathBase.Value` e.g to prevent double encoding. But here we are updating `window.location.origin` directly, that maybe needs to be correctly escaped.

If so we would need to use `PathBase.ToUriComponent()` or `PathBase.ToString()` that does the same

        /// <summary>
        /// Provides the path string escaped in a way which is correct for combining into the URI representation.
        /// </summary>
        /// <returns>The escaped path value</returns>
        public override string ToString()
        {
            return ToUriComponent();
        }